packages/redux-action-router/src/urlchangeSupport.js

```diff
@@ -0,0 +1,81 @@
+/* global window */
+import urlMapper from 'url-mapper';
+
+function upgradeHistory() {
+  window.history.pushState = (f => function pushState() {
+    const ret = f.apply(this, arguments);
+    window.dispatchEvent(new Event('pushState'));
+    window.dispatchEvent(new Event('locationchange'));
+    return ret;
+  })(window.history.pushState);
+
+  window.history.replaceState = (f => function replaceState() {
+    const ret = f.apply(this, arguments);
+    window.dispatchEvent(new Event('replaceState'));
+    window.dispatchEvent(new Event('locationchange'));
+    return ret;
+  })(window.history.replaceState);
+}
+
+let isHistoryUpgraded = false;
+export default function init(onChange, urlActionMap) {
+
+  if (!isHistoryUpgraded) {
+    upgradeHistory();
+    isHistoryUpgraded = true;
+  }
+
+  window.addEventListener('popstate', onPopState);
+  window.addEventListener('locationchange', onLocationChange);
+
+  function onLocationChange() {
+    processUrl();
+  }
+
+  function onPopState() {
+    window.dispatchEvent(new Event('locationchange'))
+  }
+
+  // TODO: Still reload page when user clicks on <a/> tag
```

Comment:
I think there are basically three choices for how to handle clicks on `<a>` tags:
1.  We could provide a React `<Link>` component that you are supposed to use to use instead of the `<a>` tag that catches the click events and converts them to `history.pushState` calls (this is what `react-router`, `reach-router`, and `react-router5` do)
2. We could explicitly state that you have to handle catching the events yourself and give you a nice function to use to do that (this is what `router5` does, if you don't use the React plugin)
3. We could wire something into the DOM that catches the click event on all `<a>` tags and convert them into `history.pushState` calls.  If we did this, it's pretty invasive, so we'd want to clearly document it and probably make it optional.

We could also do more than one of these.  1 and 3 might to both have as options, if they are both "opt in".